### ISSUE TITLE
Add possibility to save charts as images programmatically

### ISSUE THREAD

**Issue by kMutagene** [plotly/Plotly.NET#21]
Creating images locally without having to open plots and click on the save as image button would be awesome. This however will take some work.

The only way i found to do this so far is using [Orca](https://github.com/plotly/orca), which is a command line utility that plotly itself uses to render images. Orca can be either installes as npm module, conda package, or deployed as a docker container. All of these add some overhead but i still think this is usefull. I will start to implement some functions to call orca functions and we will see how it goes.

**Comment by sanchez** [comment]
It is overkill but it would be possible to use [Puppeteer-Sharp](https://github.com/hardkoded/puppeteer-sharp) to render the page and then either inject JavaScript and take the screenshot or use Puppeteer to take the screenshot on page render finish

**Comment by kMutagene** [comment]
On the first look, this actually seems to be less overhead than using Orca in a docker container, thanks for the tip!

**Comment by kMutagene** [comment]
thanks again @sanchez , i had a short go at it and it instantly worked. Here is a POC gist for future reference:

https://gist.github.com/kMutagene/afadbd5b9cb8e6b0401e94338a2644e4 

the rendered chart:

![PuppeteerPlot](https://user-images.githubusercontent.com/21338071/82787446-b3e1e480-9e66-11ea-8f25-205e1f275084.png)

**Comment by sanchez** [comment]
no worries, excited to see this feature incorporated!

idk if plotly will allow you to but you might want to wait for a response from plotly to confirm that the chart has been rendered and the page is done. A dirty way would just be a fixed delay as well

**Comment by kMutagene** [comment]
The way to go here is providing a [Kaleido](https://github.com/plotly/Kaleido) interface to render Plotly.NET Charts as images.

**Comment by nhirschey** [comment]
I'd like to register my interest in this feature so that I can easily include F#-generated figures in latex documents.

**Comment by ilyalatt** [comment]
I've published a package that uses PuppeteerSharp and Chromium to render charts.
It's called [Plotly.NET.PuppeteerRenderer](https://github.com/ilyalatt/Plotly.NET.PuppeteerRenderer).
The code is written in C# but it should not be a problem to use it in F#.
Here's a snippet:

```csharp
var chart = Chart.Line<int, int, int, int>(
    x: new[] { 1, 2, 3 },
    y: new[] { 4, 5, 6 }
);

await using var renderer = await PlotlyRenderer.FetchBrowserAndLaunch();
var png = await renderer.Render(1024, 768, chart.ToFullScreenHtml());
await File.WriteAllBytesAsync("chart.png", png);
```

**Comment by kMutagene** [comment]
@ilyalatt thanks for that!

**Comment by kMutagene** [comment]
closed via #94. The docs for the new `Plotly.NET.ImageExport` package can be found [here](https://plotly.net/0_1_image-export.html)
